cmd/storage-provisioner/main.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"k8s.io/minikube/cmd/localkube/cmd"
+	"sync"
+)
+
+func main() {
+	localkubeServer := cmd.NewLocalkubeServer()
+	storageProvisionerServer := localkubeServer.NewStorageProvisionerServer()
```

Comment:
main should look something like
```
	flag.Parse()
	if err := localkube.StartStorageProvisioner(); err != nil {
		glog.Exit(err)
	}
```

Here, instead of creating a new localkube server, change the signature to be `func StartStorageProvisioner() error` and call that directly.

https://github.com/kubernetes/minikube/blob/6a53c0c06c90b9b718ed17f026fd2dfe1af9f785/pkg/localkube/storage_provisioner.go#L117

Reply:
SGTM, but we don't need the flag.Parse() in this case, right?

Reply:
glog (the logging library) adds some flags to set verbosity, which would be nice to have on this binary (you might want to run the container command as `storage-provisioner --v 10 --logtostderr`)

Unfortunately, some of the localkube dependencies also register their flags, so theres some extra crap that gets added, but thats fine.